drone-fly-integration-tests/src/test/java/com/expediagroup/dataplatform/dronefly/core/integration/DroneFlyIntegrationTest.java

```diff
@@ -0,0 +1,194 @@
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.metastore.HiveMetaStore.HMSHandler;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.events.AddPartitionEvent;
+import org.apache.hadoop.hive.metastore.events.CreateTableEvent;
+import org.apache.hadoop.hive.metastore.events.ListenerEvent;
+import org.apache.hadoop.hive.metastore.messaging.EventMessage.EventType;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.awaitility.Duration;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.listener.ContainerProperties;
+import org.springframework.kafka.listener.KafkaMessageListenerContainer;
+import org.springframework.kafka.listener.MessageListener;
+import org.springframework.kafka.test.EmbeddedKafkaBroker;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.kafka.test.utils.ContainerTestUtils;
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import com.google.common.collect.Lists;
+
+import com.expediagroup.apiary.extensions.events.metastore.kafka.listener.KafkaMetaStoreEventListener;
+import com.expediagroup.dataplatform.dronefly.app.DroneFly;
+
+@EmbeddedKafka(count = 1, controlledShutdown = true, topics = { TOPIC }, partitions = 1)
+@ExtendWith(SpringExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class DroneFlyIntegrationTest {
+
+  private @Mock HMSHandler hmsHandler;
+
+  protected final ExecutorService executorService = Executors.newFixedThreadPool(1);
+  private static Configuration CONF = new Configuration();
+
+  private KafkaMetaStoreEventListener kafkaMetaStoreEventListener;
+
+  @Autowired
+  private EmbeddedKafkaBroker embeddedKafkaBroker;
+
+  private BlockingQueue<ConsumerRecord<String, String>> records;
+
+  private KafkaMessageListenerContainer<String, String> container;
+
+  @BeforeAll
+  void setUp() throws InterruptedException {
+    /**
+     * The function initEmbeddedKafa() is required so that EmbeddedKafka waits for the consumer group assignment to
+     * complete.
+     * https://stackoverflow.com/questions/47312373/embeddedkafka-sending-messages-to-consumer-after-delay-in-subsequent-test
+     */
+    initEmbeddedKafa();
+    System.setProperty("instance.name", "test");
```

Comment:
Do these need to be cleared in the @AfterAll so they don't impact other tests?

Reply:
There are no other tests at the moment. These properties are basically used by the Drone Fly Spring Boot app. and that is only killed after all tests are going to be complete. So, I don't think we need to reset all of them at the moment. Perhaps, when I add more tests later, it might be required for using another listener.

Reply:
OK.